language/tools/move-cli/src/base/movey_upload.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::utils::{credential, credential::TestMode};
+use anyhow::bail;
+use clap::*;
+use move_command_line_common::movey;
+use move_package::BuildConfig;
+use reqwest::blocking::Client;
+use std::{fs, path::PathBuf, process::Command};
+
+#[derive(serde::Serialize, Default)]
+pub struct MoveyUploadRequest {
+    github_repo_url: String,
+    total_files: usize,
+    token: String,
+    subdir: String,
+}
+
+/// Upload the package to Movey.net.
+#[derive(Parser)]
+#[clap(name = "movey-upload")]
+pub struct MoveyUpload {
+    #[clap(long = "test-path")]
+    test_path: Option<String>,
+}
+
+impl MoveyUpload {
+    pub fn execute(self, _path: Option<PathBuf>, config: BuildConfig) -> anyhow::Result<()> {
+        let mut movey_upload_request: MoveyUploadRequest = Default::default();
+        let mut output = Command::new("git")
+            .current_dir(".")
+            .args(&["remote", "-v"])
+            .output()
+            .unwrap();
+        if !output.status.success() || output.stdout.is_empty() {
+            bail!("invalid git repository")
+        }
+
+        let lines = String::from_utf8_lossy(output.stdout.as_slice());
+        let lines = lines.split("\n");
+        for line in lines {
+            if line.contains("github.com") {
+                let tokens: Vec<&str> = line.split(&['\t', ' '][..]).collect();
+                if tokens.len() != 3 {
+                    bail!("invalid remote url")
+                }
+                // convert ssh url to https
+                if tokens[1].starts_with("git@github.com") {
+                    let https_url = tokens[1]
+                        .replace(":", "/")
+                        .replace("git@", "https://")
+                        .replace(".git", "");
+                    movey_upload_request.github_repo_url = https_url;
+                    break;
+                }
+                movey_upload_request.github_repo_url = String::from(tokens[1]);
+                break;
+            }
+        }
+
+        output = Command::new("git")
+            .current_dir(".")
+            .args(&["rev-parse", "--show-prefix"])
+            .output()
+            .unwrap();
+        let subdir = String::from_utf8_lossy(output.stdout.as_slice());
+        movey_upload_request.subdir = String::from(subdir);
+
+        output = Command::new("git")
+            .current_dir(".")
+            .args(&["ls-files"])
+            .output()
+            .unwrap();
+        let tracked_files = String::from_utf8_lossy(output.stdout.as_slice());
+        let tracked_files: Vec<&str> = tracked_files.split("\n").collect();
+        let mut total_files = tracked_files.len();
+        for file_path in tracked_files {
+            if file_path.is_empty() {
+                total_files -= 1;
+                continue;
+            }
+        }
+        movey_upload_request.total_files = total_files;
+        movey_upload_request.token = if config.test_mode {
```

Comment:
Some comments explaining the differences in behavior between test and normal operation here would be very useful.

Reply:
We have added some comments, however, we are not very sure about the wording. If you could help us write a more precise comment, it would be very appreciated.